docs/HYPERVISORS.md

```diff
@@ -0,0 +1,85 @@
+# EVE's use of hypervisors and hardware assisted virtualization
+
+EVE is slightly unusual in how it looks at hypervisors and hardware assisted virtualization technology. Unlike some of the other systems that were clearly built as virtualization platforms (ESXi, Hyper-V, etc.) EVE's ultimate task is to be able to run [Edge Containers](ECOS.md) as efficiently (and securely!) as possible. A running ECO is represented as a series of tasks consuming resources provided by each Edge Node (such as CPU, RAM, volumes, network instances, etc.). Tasks can further improve their performance (and security) by requesting accelerator and/or isolation services typically provided by type-1 or type-2 hypervisors and backed by hardware assisted virtualization.
```

Comment:
Is this preamble necessary? It seems to take away from the strength of EVE's statement of purpose. We could do the following:

* start with the second sentence: "EVE's task is to run Edge Containers as efficiently and securely as possible." Make that the intro paragraph, and it is strong, everything else follows.
* potentially change the doc name from "hypervisors" to "isolation" (or "acceleration"). As the doc makes clear, it isn't _about_ hypervisors or virtualization, which are just technologies; it is about what we want to get done: run workloads (i.e. ECOs) efficiently and securely.

Reply:
> A running ECO is represented as a series of tasks consuming resources....

"series of tasks"? Isn't an ECO as single workload, and a series of tasks is multiple ECOs?

Reply:
> Tasks can further improve their performance (and security)

No need for "and security" to be in brackets; security is a first class citizen here.

Reply:
> accelerator and/or isolation services typically provided by type-1 or type-2 hypervisors and backed by hardware assisted virtualization

I wouldn't get too deeply into the type-1 vs type-2, perhaps just say, "typically provided by hypervisors and backed by hardware assisted virtualization"

Do we want to add OS-level virtualization, i.e. container isolation, to this list?

Reply:
> Is this preamble necessary?

I like it since it allows to refocus reader's attention right away.

> "series of tasks"? Isn't an ECO as single workload, and a series of tasks is multiple ECOs?

Yup -- that's the way our new design is evolving

> No need for "and security" to be in brackets; security is a first class citizen here.

FIxed